src/diff/index.js

```diff
@@ -406,7 +406,8 @@ export function unmount(vnode, parentVNode, skipRemove) {
 	if (options.unmount) options.unmount(vnode);
 
 	if ((r = vnode.ref)) {
-		applyRef(r, null, parentVNode);
+		if (!vnode.ref.current || vnode.ref.current === vnode._dom)
+			applyRef(r, null, parentVNode);
```

Comment:
It seems like we can either drop the `r` assignment altogether, or use it for L409 as well?

Also - it seems like we can skip nulling the ref if it's already null? (when `!vnode.ref.current`)

Reply:
Do you think we should move to assigning `refs` during `commit` to also fix this for functional refs or would you prefer only fixing objects (the former will add a lot more bytes)?

Reply:
I think for now let's keep this in-place and normalize the objects - if there's not a reason to avoid using `r` here we can just use `r`. Longer term this seems like a potential case for the queue, but we'll need to check what the timing requirements are for that - eg, should refs be populated/called prior to effect hooks, or after?

Reply:
I applied your changes